### DISCUSSION TITLE
Prometheus Metrics

### DISCUSSION THREAD

**Discussion by DuvelCorp** [transmission/transmission#5719] [General]
Hi

Atm I am using this https://github.com/metalmatze/transmission-exporter to export Transmission metrics into Prometheus.
Its not maintained anymore and becomes quite old  (2019)

I searched a bit and found a few other things like this https://github.com/sandrotosi/simple-transmission-exporter and that https://github.com/leishi1313/downloader-exporter but they also look like abandoned projects.

I am wondering if I missed something else, or to say it otherwise if someone has a better transmission metrics exporter to propose ;)

Cheers

**Comment by dantearmok** [comment]
Old and possibly abandoned isn't necessarily bad, especially if it's working!

Do you have a link for Prometheus?  I've never heard of it before.

**Reply by DuvelCorp** [reply]
[Prometheus](https://prometheus.io/) is the metrics "ingestor" that works with [Grafana](https://grafana.com/). Its purpose is to collect and index all metrics from the servers and applications you want. Since transmission doesn't expose a metric API natively, you need another "tool" to export them. The tool I use atm can only understand the V2 of transmission (and so its logs/metrics), so its not a good thing in this case (I run V4 obviously).

[Here](https://grafana.com/grafana/dashboards/15116-qbittorrent-dashboard/) you have a good example of what you can do with a decent metrics exporter (with qbittorrent and not transmission)

![image](https://github.com/transmission/transmission/assets/33682764/3b086c0d-3543-4bfd-bd67-ef6853573ea2)

![image](https://github.com/transmission/transmission/assets/33682764/bd07a629-68d7-4373-8741-bd321c5671cd)

Isn't that shinning and helpful ?

And here's how my actual transmission metrics looks like and it is extremely poor compared to what other torrent clients exporters can do.

```
# HELP process_max_fds Maximum number of open file descriptors.
# TYPE process_max_fds gauge
process_max_fds 1.048576e+06
# HELP process_open_fds Number of open file descriptors.
# TYPE process_open_fds gauge
process_open_fds 11
# HELP process_resident_memory_bytes Resident memory size in bytes.
# TYPE process_resident_memory_bytes gauge
process_resident_memory_bytes 1.128448e+07
# HELP process_start_time_seconds Start time of the process since unix epoch in seconds.
# TYPE process_start_time_seconds gauge
process_start_time_seconds 1.6884031435e+09
# HELP process_virtual_memory_bytes Virtual memory size in bytes.
# TYPE process_virtual_memory_bytes gauge
process_virtual_memory_bytes 1.13737728e+08
# HELP process_virtual_memory_max_bytes Maximum amount of virtual memory available in bytes.
# TYPE process_virtual_memory_max_bytes gauge
process_virtual_memory_max_bytes -1
# HELP transmission_alt_speed_down Alternative max global download speed
# TYPE transmission_alt_speed_down gauge
transmission_alt_speed_down{enabled="0"} 50
# HELP transmission_alt_speed_up Alternative max global upload speed
# TYPE transmission_alt_speed_up gauge
transmission_alt_speed_up{enabled="0"} 50
# HELP transmission_cache_size_bytes Maximum size of the disk cache
# TYPE transmission_cache_size_bytes gauge
transmission_cache_size_bytes 4.194304e+06
# HELP transmission_free_space Free space left on device to download to
# TYPE transmission_free_space gauge
transmission_free_space{download_dir="/data/completed",incomplete_dir="/data/incomplete"} 5.59157276672e+11
# HELP transmission_global_peer_limit Maximum global number of peers
# TYPE transmission_global_peer_limit gauge
transmission_global_peer_limit 300
# HELP transmission_queue_down Max number of torrents to download at once
# TYPE transmission_queue_down gauge
transmission_queue_down{enabled="0"} 20
# HELP transmission_queue_up Max number of torrents to upload at once
# TYPE transmission_queue_up gauge
transmission_queue_up{enabled="0"} 10
# HELP transmission_seed_ratio_limit The default seed ratio for torrents to use
# TYPE transmission_seed_ratio_limit gauge
transmission_seed_ratio_limit{enabled="0"} 10
# HELP transmission_session_stats_active The time transmission is active since
# TYPE transmission_session_stats_active gauge
transmission_session_stats_active{type="cumulative"} 1.641793928e+09
transmission_session_stats_active{type="current"} 1.688389037e+09
# HELP transmission_session_stats_download_speed_bytes Current download speed in bytes
# TYPE transmission_session_stats_download_speed_bytes gauge
transmission_session_stats_download_speed_bytes 0
# HELP transmission_session_stats_downloaded_bytes The number of downloaded bytes
# TYPE transmission_session_stats_downloaded_bytes gauge
transmission_session_stats_downloaded_bytes{type="cumulative"} 6.677581990226e+12
transmission_session_stats_downloaded_bytes{type="current"} 2.20430336e+08
# HELP transmission_session_stats_files_added The number of files added
# TYPE transmission_session_stats_files_added gauge
transmission_session_stats_files_added{type="cumulative"} 273545
transmission_session_stats_files_added{type="current"} 2
```

**Comment by dantearmok** [comment]
ooOoh! coolbeans!  Although, such metrics to me personally would be depressing - I do NOT want a record of how much time I spend in Transmission!  LOL

From what I can tell, the first exporter you cite is fairly up-to-date, given the current state of Transmission.

It would be interesting to identify the additional metrics you need from Transmission, then generate an Issue and a related PR to have those items saved / exposed.

**Comment by darmach** [comment]
@DuvelCorp What's your recent experieces with prom, transmissions and exporting metrics? Is that exporter https://github.com/metalmatze/transmission-exporter optimal? I also found this https://github.com/sandrotosi/simple-transmission-exporter

Pro of the latter is prepared grafana dashboard for people who want something working out of the box: https://grafana.com/grafana/dashboards/13265-transmission-by-simple-exporter/

**Comment by DuvelCorp** [comment]
I stopped transmission and moved to qbittorent.
Now I got proper metrics and a lot more other features :-)

**Comment by fede843** [comment]
no news for this yet? it'd be really nice to get some metrics out of the box for this project
